### ISSUE TITLE
[Compatability] Dynamic Trees

### ISSUE THREAD

**Issue by SpeakerHornet** [orangeadam3/terra121#3]
It would be cool to see Dynamic Trees compatability in Terra. It would add a lot to vegetation. Here's the mod link: https://www.curseforge.com/minecraft/mc-mods/dynamictrees

**Comment by orangeadam3** [comment]
Not a bad idea. It's actually been suggested before. I can look into it.

**Comment by Barteks2x** [comment]
There is a compatibility mod for dynamic trees already, but I'm not sure if it still works with latest versions and how it interacts with this mod

**Comment by SpeakerHornet** [comment]
I tried the compatibility mod and it doesn't work.

**Comment by orangeadam3** [comment]
Does it crash or just do normal trees?

**Comment by SpeakerHornet** [comment]
It just makes normal trees

**Comment by orangeadam3** [comment]
So I tried writing some compatibility code and it sort of works. The problem is that dynamictrees uses something called a Poisson disk to keep the trees from overlaping. I really don't understand how it works so until I do compatbility will probably have to be on hold

**Comment by Barteks2x** [comment]
It should work with the existing addon if you post the right decorstion event and respect cancelling it

**Comment by orangeadam3** [comment]
I don't think that will work because I use a custom tree count that is based on the tree cover dataset and not based on the biome. I think the way this generates trees is procedural and based on the biome. When I tried compatibility I made it get the species from the biome and manually place the species. The problem is the trees overlap with each other so I'd have to figure out how to use dynamictrees' Poisson disc noise system.

**Comment by Atopik2** [comment]
What about Oh The Biomes You'll Go?
